### PR TITLE
Add log function for templates

### PR DESCRIPTION
**Pull request by artem-sidorenko**
It might be very useful for building nice tag clouds

Example:

Tag cloud built without log:

untitled

Tag cloud built with log:

untitled1

### CODE REVIEW

**Comment by digitalcraftsman** [comment]
Go's `math` package names the function for the natural logarithm (log_e) just `Log`. You use this name for the template func as well. However, users might find this name ambiguous because I would rather expect that `Log` returns log_10 instead of log_e.

Hence I would prefer to name the template function `ln` (for logarithm naturalis).

Let's take this a step further. log_e might be limiting in some cases and log_10 might be needed instead. While the `math` package has a function for this too it only adds just another function to the namespace.

Hence I suggest the addition (or even as a replacement) of a more generic logarithm template func where users can set the base as a parameter. The `math` package doesn't provide a function that takes the base as a parameter. 

But it's trivial to obtain the result in the wished by with log_10. Let's say I want to calculate log_2(8). This can be written as log_10(8) / log_10(2).

So implementing a generic function should be trivial.

File: tpl/math/math.go [hunk 1]
```diff
@@ -34,6 +35,22 @@ func (ns *Namespace) Div(a, b interface{}) (interface{}, error) {
 	return DoArithmetic(a, b, '/')
 }
 
+func (ns *Namespace) Log(a interface{}) (float64, error) {
+	av := reflect.ValueOf(a)
+	var af float64
+
+	switch av.Kind() {
```
Comment (bep) [inline tpl/math/math.go]: Use `cast. ToFloat64`.

Comment (moorereason) [inline tpl/math/math.go]: Or `cast.ToFloat64E`

**Comment by moorereason** [comment]
@digitalcraftsman,
The [Go math package](https://golang.org/pkg/math/) offers `Log`, `Log10`, `Log1p`, `Log2`, and `Logb`.  We should follow their lead as to what `Log` means.  I'm not in favor of adding all of these funcs to Hugo until someone needs them, though.

Also, I don't like `ln`.  That's too short (saves one character and loses most of its intuitiveness).  Makes me think of creating symlinks.  If I were searching for a logarithm func, I'd be looking for "log" not "ln."

**If** we put these in the global funcmap, I say we simply use the same names:  `log`, `log10`, `log1p`, `log2`, and `logb`.

Creating a single, multipurpose `log` func feels like over-engineering.

**Comment by bep** [comment]
>I'd be looking for "log" not "ln."

Without any context, I would say that `log` does some debug logging or something. I would say that 

1. Follow the names from `math.Log` etc.
2. No aliases

**Comment by digitalcraftsman** [comment]
> I'm not in favor of adding all of these funcs to Hugo until someone needs them, though.

Hence the generic func which is neither difficult to use nor to implement.

> Also, I don't like ln. That's too short (saves one character and loses most of its intuitiveness). Makes me think of creating symlinks. If I were searching for a logarithm func, I'd be looking for "log" not "ln."

Well, `ln` is the button you'll find on a calculator.

File: tpl/math/math.go (hunk 1 above)
Comment (artem-sidorenko) [inline tpl/math/math.go]: @bep @moorereason  done, thanks

**Comment by artem-sidorenko** [comment]
@moorereason done

**Pull request merged**